blockmanager.go

```diff
@@ -2383,6 +2383,33 @@ func (b *blockManager) handleHeadersMsg(hmsg *headersMsg) {
 	// atomically in order to improve peformance.
 	headerWriteBatch := make([]headerfs.BlockHeader, 0, len(msg.Headers))
 
+	// Explicitly check that each header in msg.Headers builds off of the
```

Comment:
Can be moved out into a some helper function. Also would have though the functions we exported from `btcd` would handle it so we don't need to do it again?